api/src/main/java/io/opentelemetry/trace/BigendianEncoding.java

```diff
@@ -165,5 +173,24 @@ private static void byteToBase16(byte value, char[] dest, int destOffset) {
     dest[destOffset + 1] = ENCODING[b | 0x100];
   }
 
+  public static boolean isValidBase16String(CharSequence value) {
+    for (int i = 0; i < value.length(); i++) {
+      char b = value.charAt(i);
+      // 48..57 && 97..102 are valid
+      if ((b < 48 || b > 57) && (b < 97 || b > 102)) {
```

Comment:
I'm pretty sure they're the same, and maybe it's just me but I find writing the same code as the comment would be more readable 

```suggestion
      if (!(b >= 48 && b <= 57) || !(b >= 97 && b <= 102)) {
```

Reply:
heh. I agree with you. However, what you pasted isn't correct (IDEA says it's always true!). I'll figure out hot to make it what you intended. :)

Reply:
check out the update. I think it's now much easier to understand.